proposals/0034-tpu-vote-using-quic.md

```diff
@@ -0,0 +1,112 @@
+---
+simd: '0034'
+title: TPU Vote using QUIC
+authors:
+  - Lijun Wang <lijun@anza.xyz>
+category: Standard
+type: Core
+status: Review
+created: 2024-11-13
+development: 
+  - Anza - WIP
+  - Firedancer - Not started
+---
+
+## Summary
+
+Use QUIC for transporting TPU votes among Solana validators. This requires
+supporing receiving QUIC based vote TPU packets on the server side and sending
+QUIC-based TPU vote packets on the client side.
+
+
+## Motivation
+
+As timely vote credits are awarded to validators, they might be incentived to
+increase the TPU vote traffic to ensure their votes are received in a timely
+manner. This could cause congestions and impact overall TPU vote processing
+effectiveness. The concurrent UDP based TPU vote does not have any flow control
+mechanism.
+
+We propose to apply the pattern taken for TPU transaction processing to TPU vote
+processing -- by utlizing the flow control mechanism which were developed
+including built-in QUIC protocol level flow control, and application-level rate
+limiting on connections and packets.
+
+## Alternatives Considered
+
+There is no readily-available alternative to QUIC which addresses some of the
+requirements such as security (reliability when applying QOS), low latency and
+flow control. We could solve the security and flow control with TLS over TCP
+the concern is with the latency and head-of-line problems. We could also
+customize and build our own rate limiting mechanism based on the UDP directly,
+this is non-trivial and cannot solve the security problem without also rely on
+some sort of crypto handshaking. 
+
+## New Terminology
+
+None
+
+## Detailed Design
+
+On the server side, the validator will bind to a new QUIC endpoint. Its
+corresponding port will be published to the network in the ContactInfo via
+Gossip. The client side will use the TPU vote QUIC port published by the server
+to connect to the server.
+
+The TPU vote will be using the same QUIC implementation used by regular
+transaction transportation. The client and server both uses their validator's
+identity key to sign the certificate which is used to validate the validator's
+identity especially on the server side for the purpose of provding QOS based on
+the client's stakes by checking the client's Pubkey -- stake weighted QOS.
```

Comment:
Just saying: validator id to vote account association and the stake distribution could change at the Epoch boundary. At Epoch boundary you probably need to accept if the validator has take in either the old Epoch or the new Epoch.

Reply:
Stake info is periodically updated and is eventual consistent. It applies the QOS based on the latest info as it see it. I do not think we need to complicate it to look at potentially two sets of data.